### PR TITLE
Implement client-side compression tests

### PR DESCRIPTION
**Pull request by ericgribkoff**
This adds the client-side compression tests for https://github.com/grpc/grpc-java/issues/3195.

The first commit brings `messages.proto` up to date with its corresponding file in grpc/grpc.

@ejona86 Filed https://github.com/grpc/grpc/pull/11739 to update the interop test descriptions to explicitly allow for implementations that cannot verify the received messages are truly compressed

### CODE REVIEW

**Comment by ericgribkoff** [comment]
@makdharma  fyi

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java [hunk 1]
```diff
@@ -319,6 +320,107 @@ public void largeUnary() throws Exception {
...
+            .build();
+
+    // Send a non-compressed message with expectCompress=true. Servers supporting this test case
+    // should return INVALID_ARGUMENT.
+    try {
+      blockingStub.unaryCall(expectCompressedRequest);
+      fail("expected INVALID_ARGUMENT");
+    } catch (StatusRuntimeException e) {
+      assertEquals(Status.INVALID_ARGUMENT.getCode(), e.getStatus().getCode());
+      assertEquals("Compressed request expectation not met.", e.getStatus().getDescription());
```
Comment (ejona86) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java]: This is overly strict and should be removed. It is not specified to check this string, and the Server-side feature CompressedRequest does not mandate a particular status description. Ditto elsewhere.

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java [hunk 2]
```diff
@@ -395,6 +497,90 @@ public void clientStreaming() throws Exception {
...
+    Throwable e = responseObserver.getError();
+    assertNotNull("expected INVALID_ARGUMENT", e);
+    assertEquals(Status.INVALID_ARGUMENT.getCode(), Status.fromThrowable(e).getCode());
+    assertEquals(
+        "Compressed request expectation not met.", Status.fromThrowable(e).getDescription());
+
+    // Start a new stream
+    responseObserver = StreamRecorder.create();
+    ClientCall<StreamingInputCallRequest, StreamingInputCallResponse> call =
+        channel.newCall(
```
Comment (ejona86) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java]: There should be no need to use `ClientCall` directly. Cast the stream observer to `ClientCallStreamObserver`, which has `setMessageCompression`.

Comment (ericgribkoff) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java]: Done

**Pull request merged**